### PR TITLE
refactor formio builder for v2

### PR DESCRIPTION
**Pull request by nh758**
need to update route for image/file uploads

Should this move to the `ABDesigner`? We would only need this custom webix component for the Designer but all our other custom components live in platform web.

### CODE REVIEW

File: webix_custom_components/formioBuilder.js [hunk 1]
```diff
@@ -64,362 +68,334 @@ module.exports = class ABCustomFormIOBuilder {
...
-                        },
-                     };
-                     break;
+   /**
+    * @method parseDataObjects
+    *
+    * @param store {webix.TreeCollection}
+    * @param elem {Object} the webix element
+    * @param parentId {integer - nullable} id of parent id
+    */
```
Comment (Hiro-Nakamura) [inline webix_custom_components/formioBuilder.js:89]: this doesn't seem to be the proper header for this function.

File: webix_custom_components/formioBuilder.js [hunk 2]
```diff
@@ -10,50 +10,54 @@ module.exports = class ABCustomFormIOBuilder {
...
-            var comp = _logic.parseDataFields(config.dataFields);
-            var formComponents = config.formComponents
+         $init: async function (config) {
+            const comp = this.parseDataFields(config.dataFields);
+            const formComponents = config.formComponents
                ? config.formComponents
-               : {};
-            Formio.builder(this.$view, formComponents, {
+               : { components: [comp.approveButton.schema] };
+            console.log(formComponents);
```
Comment (Hiro-Nakamura) [inline webix_custom_components/formioBuilder.js:48]: let's not leave `console.log()` statements in here.

File: webix_custom_components/formioBuilder.js [hunk 3]
```diff
@@ -10,50 +10,54 @@ module.exports = class ABCustomFormIOBuilder {
...
+            /* global Formio */
+            this.builder = await Formio.builder(this.$view, formComponents, {
```
Comment (Hiro-Nakamura) [inline webix_custom_components/formioBuilder.js:50]: we should probably wrap this in a `try ... catch ...`

File: webix_custom_components/formioBuilder.js [hunk 4]
```diff
@@ -64,362 +68,334 @@ module.exports = class ABCustomFormIOBuilder {
...
+    * @param store {webix.TreeCollection}
+    * @param elem {Object} the webix element
+    * @param parentId {integer - nullable} id of parent id
+    */
+   parseDataFields(fields) {
+      const components = {};
+      // objects.forEach((obj) => {
+      //     var fields = obj.fields();
+      //     console.log(fields);
+      fields.forEach((entry) => {
```
Comment (Hiro-Nakamura) [inline webix_custom_components/formioBuilder.js:95]: so what are these `fields` here?  are they instances of an `ABField` objects?  or are they another kind of json definition?  (I don't see us using any method on them)

**Review by Hiro-Nakamura — CHANGES_REQUESTED** [review]

**Comment by Hiro-Nakamura** [comment]
I agree, I think this code is only used in the ABDesigner, so we should move it there. 

The other custom components we define should be components we use in the normal operation of the AppBuilder.  But for things only used in the ABDesigner, we should move them there.

As for the routes, I keep thinking we should ask an `ABFieldFile` or `ABFieldImage` for it's route and use that there.  That way the routes always change if we update the `official` place for them.  But I'm not sure what we have available in this function at the time.  So we can keep it as is if we don't have access to them.

**Comment by nh758** [comment]
Thanks Johnny, I'll look moving this to the Designer and getting the route from the fields.

File: AppBuilder/platform/dataFields/ABFieldFile.js [hunk 5]
```diff
@@ -48,8 +48,7 @@ module.exports = class ABFieldFile extends ABFieldFileCore {
                   // TODO: a reminder that you still got alot on the server to do!
                   webix.alert({
                      title: "!! TODO !!",
-                     text:
-                        "Tell a Developer to actually pay attention to this!",
+                     text: "Tell a Developer to actually pay attention to this!",
```
Comment (Hiro-Nakamura) [inline AppBuilder/platform/dataFields/ABFieldFile.js:51]: I mean ... I'm thinking that at some point we actually need to pay attention to this.

**Review by Hiro-Nakamura — APPROVED** [review]
lgtm

**Pull request merged**